src/mod2c_core/nocpout.c

```diff
@@ -1653,6 +1654,52 @@ static void _destructor(Prop* _prop) {\n\
 	}
 }
 
+static void add_readion(List* read_variables, const char* name) {
+	Symbol* const sym = install(name, NAME);
+    sym->nrntype |= IONCONC;
+    sym->nrntype |= IONCONC_IMPLICIT;
+    lappendsym(read_variables, sym);
+}
+static void check_sufficient_ion_read_statements(char* ion_name,
```

Comment:
to clarify : when I read this function name, I got a bit puzzled thinking what here we meant by "**sufficient** ion read statements"?
If I am not mistaken, we are iterating via both read and write variables. So "sufficient" confused me a bit.

Reply:
I think I agree with you that it's not a great name... (naming is hard!)

The point, IIRC, is that we need to have(/implicitly add) a `USEION` statement for **both** `IONNAMEo` and `IONNAMEi` jf we have a real `USEION` statement for either (I didn't re-check just now, but I think that's the condition for emitting a call to `nrn_wrote_conc`). If one of them is missing, we implicitly add it as a `READ`.

We could just drop `_read` from the function name?
Or: `add_implicit_ion_read_statements`?

Reply:
> Or: add_implicit_ion_read_statements?

sounds a bit better to me but either is fine!